src/neptune/metadata_containers/run.py

```diff
@@ -99,6 +99,9 @@
 from neptune.types.atoms.git_ref import GitRefDisabled
 from neptune.types.mode import Mode
 
+if TYPE_CHECKING:
```

Comment:
Is there any specific reason why import code that are related to signals wrapped by TYPE_CHECKING?

Reply:
In general, this solves circular-imports when sometimes we're trying to import some parent objects just for mypy/typing requirements. Even if that's not a case it's a part of my personal practice to indicate that specific imports are just for typing and should be only triggered whenever `mypy` is processing the codebase.